### PR TITLE
v1.6.3

### PR DESCRIPTION
**Pull request by zomars**
- #2486
- #2506
- #2553
- #2602
- #2652
- #2669
- #2671
- #2674
- #2696
- #2707
- #2718
- #2719
- #2722
- #2723
- #2725
- #2727
- #2728
- #2729
- #2730
- #2732
- #2737
- #2743
- #2745
- #2747
- #2749
- #2755
- #2756
- #2759
- #2760
- #2761
- #2764
- #2768
- #2769
- #2770
- #2773
- #2777
- #2784
- #2790
- #2906

### CODE REVIEW

File: apps/web/components/App.tsx [hunk 1]
```diff
@@ -70,9 +73,13 @@ export default function App({
...
             "Content-Type": "application/json",
           },
+        }).then((data) => {
+          setIsLoading(false);
+          return data;
         });
         if (result.status === 200) {
-          setInstalledApp(true);
+          const res = await result.json();
+          setInstalledApp(res.count);
```
Comment (alannnc) [inline apps/web/components/App.tsx:82]:
```suggestion
          setInstalledAppCount(res.count);
```

File: apps/web/components/App.tsx [hunk 2]
```diff
@@ -107,19 +114,35 @@ export default function App({
...
-                  <InstallAppButton
-                    type={type}
-                    render={(buttonProps) => (
-                      <Button data-testid="install-app-button" {...buttonProps}>
-                        {t("install_app")}
+                {!isLoading ? (
+                  isGlobal || installedApp > 0 ? (
+                    <div className="space-x-3">
+                      <Button StartIcon={CheckIcon} color="secondary" disabled>
+                        {installedApp > 0
```
Comment (alannnc) [inline apps/web/components/App.tsx:121]:
```suggestion
                        {installedAppCount > 0
```

File: apps/web/pages/api/book/confirm.ts [hunk 3]
```diff
@@ -74,230 +86,228 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
+  const evt: CalendarEvent = {
+    type: booking.title,
+    title: booking.title,
+    description: booking.description,
+    customInputs: isPrismaObjOrUndefined(booking.customInputs),
+    startTime: booking.startTime.toISOString(),
+    endTime: booking.endTime.toISOString(),
+    organizer: {
+      email: currentUser.email,
+      name: currentUser.name || "Unnamed",
```
Comment (alannnc) [inline apps/web/pages/api/book/confirm.ts:172]: Unnamed or Nameless

Comment (zomars) [reply apps/web/pages/api/book/confirm.ts:172]: Feel free to commit it

File: apps/web/pages/api/book/confirm.ts [hunk 4]
```diff
@@ -74,230 +86,228 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
+        errorCode: "BookingCreatingMeetingFailed",
+        message: "Booking failed",
+      };
 
-        log.error(`Booking ${currentUser.username} failed`, error, results);
-      } else {
-        const metadata: AdditionInformation = {};
+      log.error(`Booking ${currentUser.username} failed`, error, results);
+    } else {
+      const metadata: AdditionInformation = {};
```
Comment (alannnc) [inline apps/web/pages/api/book/confirm.ts:211]:
```suggestion
      const metadata: AdditionalInformation = {};
```

File: apps/web/pages/api/book/confirm.ts [hunk 5]
```diff
@@ -74,230 +86,228 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
-        }
+      if (results.length) {
+        // TODO: Handle created event metadata more elegantly
+        metadata.hangoutLink = results[0].createdEvent?.hangoutLink;
+        metadata.conferenceData = results[0].createdEvent?.conferenceData;
+        metadata.entryPoints = results[0].createdEvent?.entryPoints;
       }
+      try {
+        await sendScheduledEmails(
+          { ...evt, additionInformation: metadata },
```
Comment (alannnc) [inline apps/web/pages/api/book/confirm.ts:221]:
```suggestion
          { ...evt, additionalInformation: metadata },
```

File: apps/web/pages/api/email.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import { NextApiRequest, NextApiResponse } from "next";
```
[the remark this answers is not in the document]
Comment (zomars) [reply apps/web/pages/api/email.ts:1]: There's a line that won't let you use this on production. Leave it for a follow up

**Review by agustif — APPROVED** [review]
Awesome work everyone, it feels like this release is all about bug fixing, making the codebase cleaner and more modular, and overall a great milestone. Also some nice new ui touches/features 💯 

image

**Pull request merged**